authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/web/services/CaasProviderIdUpdateService.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 sovity GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *      sovity GmbH - initial implementation
+ */
+
+package de.sovity.authorityportal.web.services
+
+import de.sovity.authorityportal.db.jooq.Tables
+import de.sovity.authorityportal.db.jooq.enums.ConnectorType
+import io.quarkus.scheduler.Scheduled
+import jakarta.enterprise.context.ApplicationScoped
+import org.jooq.DSLContext
+
+@ApplicationScoped
+class CaasProviderIdUpdateService(
+    val dsl: DSLContext,
+    val organizationService: OrganizationService
+) {
+
+    @Scheduled(every = "24h")
+    fun updateMissingProviderIds() {
```

Comment:
so does this really have to be done every 24h or do new connectors now automatically get the providerId set and we would only need this executed once?

Reply:
New connectors get the providerId set only if sovity is already registered, as soon as sovity registers, this could be executed only once (by firing an event or so I guess).
But consider this:

- Delete the organization sovity
- Request a CaaS
- Re-invite sovity much later

Then the connector would forever lack the providerId. It's a very stupid edge-case but the implemented method would prevent it